test/evaluation/check-scenario-1.spec.ts

```diff
@@ -192,7 +192,7 @@ describe('1st scenario', () => {
      */
     const selectFromResult = pejs.selectFrom(pd, holderWallet.verifiableCredentials, [holderWallet.holder]);
     expect(selectFromResult.matches?.length).toEqual(1);
-    expect(selectFromResult.matches).toEqual([{ rule: 'all', matches: ['$[2]'] }]);
+    expect(selectFromResult.matches).toEqual([{ rule: 'all', matches: ['$[0]'], name: undefined }]);
```

Comment:
name should not be undefined.

In method 

```
private matchWithoutSubmissionRequirements(
    marked: HandlerCheckResult[],
    pd: PresentationDefinition
  ): SubmissionRequirementMatch[] {
```
Line 125.
`const partitionedResults: Map<string, string[]> = this.partitionCheckResults(marked);`

`marked [index]` contains field input_descriptor_path having value `$.input_descriptors[0]`
Use this to extract the name and fill in such that line 30 has the name correctly filled.

Reply:
Fixed!

Reply:
Removing it is not fixing it. It should always have a name. Either it will be a id/name of the SubmissionRequirements or it will be the id/name of the InputDescriptor.

Reply:
Name is optional in the input descriptor and submission requirement, so it cannot be used. I'm using the id property of the input descriptors, therefore submission requirement match is not considered duplicated anymore and the expected result of a test case had to be changed.